apps/web/src/pages/Pools/PoolLayout.tsx

```diff
@@ -64,13 +64,32 @@ const PoolHeader = ({ poolQuery }: PoolHeaderProps) => {
   const pool = getFragment(PoolLayout_Fragment, poolQuery);
 
   const pages = useAdminPoolPages(intl, pool);
-
-  const poolTitle = getShortPoolTitleLabel(intl, pool);
   const currentPage = useCurrentPage<PageNavKeys>(pages);
+
   const subTitle = pool.team
     ? getLocalizedName(pool.team?.displayName, intl)
     : currentPage?.subtitle;
 
+  const heroTitle = () => {
+    if (currentPage?.link.url.includes("edit")) {
+      return currentPage?.title;
+    }
+    if (currentPage?.link.url.includes("plan")) {
+      return currentPage?.title;
+    }
+    return getShortPoolTitleLabel(intl, pool);
+  };
+
+  const heroSubtitle = () => {
+    if (currentPage?.link.url.includes("edit")) {
+      return currentPage?.subtitle;
+    }
+    if (currentPage?.link.url.includes("plan")) {
+      return currentPage?.subtitle;
+    }
+    return subTitle;
+  };
+
```

Comment:
Can you change this to a variable? That way it is just a reference later on and we don;t need to execute the function every time we need to check it.

```tsx
  let heroTitle: string | undefined = getShortPoolTitleLabel(intl, pool);
  if (currentPage?.link.url.includes("edit")) {
    heroTitle = currentPage?.title;
  }
  if (currentPage?.link.url.includes("plan")) {
    heroTitle = currentPage?.title;
  }

  let heroSubtitle: string | undefined = subTitle;
  if (currentPage?.link.url.includes("edit")) {
    heroSubtitle = currentPage?.subtitle;
  }
  if (currentPage?.link.url.includes("plan")) {
    heroSubtitle = currentPage?.subtitle;
  }
```

Reply:
I actually followed an anti-pattern from the beginning defining a component within another component (@esizer has created an [issue](https://github.com/GCTC-NTGC/gc-digital-talent/issues/10228) for this for existing code). I've refactored it so that the functions are outside the component, please see: ca5d06ebd9f2dc4ad7e2cbb4e086922938b39031

Reply:
Looks good!